core/trino-main/src/test/java/io/trino/server/security/TestResourceSecurity.java

```diff
@@ -334,11 +343,16 @@ public void testJwtAuthenticator()
             assertAuthenticationDisabled(httpServerInfo.getHttpUri());
 
             String hmac = Files.readString(Paths.get(HMAC_KEY));
-            String token = Jwts.builder()
+            JwtBuilder tokenBuilder = Jwts.builder()
                     .signWith(SignatureAlgorithm.HS256, hmac)
-                    .setSubject("test-user")
-                    .setExpiration(Date.from(ZonedDateTime.now().plusMinutes(5).toInstant()))
-                    .compact();
+                    .setExpiration(Date.from(ZonedDateTime.now().plusMinutes(5).toInstant()));
+            if (principalField.isPresent()) {
```

Comment:
This could use the new method
```java
principalField.ifPresentOrElse(
        field -> tokenBuilder.claim(field, "test-user"),
        () -> tokenBuilder.setSubject("test-user"));